src/pages/signin/SignInPageLayout/index.js

```diff
@@ -1,51 +1,22 @@
 import React from 'react';
-import {Dimensions} from 'react-native';
-import _ from 'underscore';
 import variables from '../../../styles/variables';
 import SignInPageLayoutNarrow from './SignInPageLayoutNarrow';
 import SignInPageLayoutWide from './SignInPageLayoutWide';
+import withWindowDimensions, {windowDimensionsPropTypes} from '../../../components/withWindowDimensions';
 
-class LoginForm extends React.Component {
-    constructor(props) {
-        super(props);
+const propTypes = {
+    ...windowDimensionsPropTypes,
+};
 
-        this.toggleScreenWidth = _.debounce(this.toggleScreenWidth.bind(this), 1000, true);
+const SignInPageLayout = props => (
+    props.windowDimensions.width > variables.mobileResponsiveWidthBreakpoint
```

Comment:
I really like the HOC. It might make sense at some point to have the HOC provide a prop called `isMobileWidth` since this logic is reused a lot and seems to be pretty common.

Reply:
Yeah I was thinking similar. Gonna maybe save this for a follow up.